README.md

```diff
@@ -45,3 +45,58 @@ You can use the command `docker compose exec django python manage.py createsuper
+
+## Feature Explanation
+
+Since this webserver is meant to control IoT devices and receive data form IoT devices, we implemented a feature where a user can create an IoT device that uploads data to the server securely. 
+
+At the current time, we have it set up so that a RPi can upload time-lapses of plant growth. The code for the webserver running on the Pi can be found at `device-uploading/time_lapse.py`. However, since we do not expect a TA to set up or have a Raspberry Pi to test this (and it would be difficult to do anyway when testing of localhost) we have included a file that can be used to simulate/spoof what the raspberry pi is doing from your own computer. This file can be found at `device_uploading/device_spoofer.py`.
+
+Additionally, to help an admin user manage these devices, we have created admin commands inside of Django that allows the admin to register, de-activate, and activate a device. They are as follows:
+- `register_relay_device`: Registers a new device into the database, sets it as active, and returns to the admin user the auth-token/password to upload for that device's authentication. 
+- `deactivate_relay_device`: Marks a device as de-activated and will reject any requests coming from this device even if it has a valid authentication token. 
+- `activate_relay_device`: Marks a device as activated and will allow and requests coming from this device when it has valid authentication credentials. 
+
+We believe both the features for the time-lapse uploads from the Pi _and_ the admin commands both count as separate valid features for the purposes of Part 3 Objective 3. Even if you do not count the Pi stuff since we cannot have you run the script on a Pi to our web server to test, the admin commands should still be sufficient to count as the additional feature. 
+
+## Testing Procedure
+
+1. Locate the file `device_spoofer.py` inside the `device_uploading` directory. Copy this to a totally new and separate location on your computer from the directory/repo for this project. For example, if you cloned this repo so that it is at `~Projects/Grading/Project-EverGreen/` then you could move this file to `~/Projects/Grading/Spoofing/`. 
+
+2. Inside of the directory in which you copied `device_spoofer.py`, also create a new directory that is named `capture`. For example, if you placed `device_spoofer.py` at `~/Projects/Grading/Spoofing/device_spoofer.py`, you should have a directory `~/Projects/Grading/Spoofing/capture/`. 
+
+3. Place at least three unique `.mp4` files that are less than 40MB in size with unique names inside of the `capture` directory you just created. These will be used for testing. It is important that they are all different files with different contents _and_ different filenames.
+
+3. Spin up the containers to launch the app locally. You can do this using the command `docker compose up --build --force-recreate` from the directory you placed the project/repo.
+
```

Comment:
You duplicated step #3 so everything else is off-by-one; though rendering normalizes to fix this in display